os/kernel/debug/mem_leak_checker.c

```diff
@@ -0,0 +1,407 @@
+/****************************************************************************
+ *
+ * Copyright 2023 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <stdlib.h>
+#include <debug.h>
+#include <stdio.h>
+#include <string.h>
+#include <queue.h>
+#include <sys/types.h>
+#include <tinyara/mm/mm.h>
+#include <tinyara/mm/heap_regioninfo.h>
+#include <binary_manager/binary_manager.h>
+#include <arch/chip/memory_region.h>
+
+#include "../binfmt/libelf/libelf.h"
```

Comment:
I think you include this to use `get_bin_addr_list` function. Then, the prototype should be in a public header. Header files in each source folder are internal headers. 
Let's select a way among two options
1. move the prototype to a public header of loader
2. add CFLAGS in this Makefile for loader path (you can refer board's Makefile for this)

Reply:
I have moved the declaration to elf.h, also there was requirement of structure from libelf.h, which is also moved to elf.h. Since the elf.h is included in libelf.h, there should not be any problems.